Sources/StreamChat/ChatClient.swift

```diff
@@ -413,6 +410,11 @@ public class _ChatClient<ExtraData: ExtraDataTypes> {
             userConnectionProvider: userConnectionProvider,
             completion: completion
         )
+        
+        backgroundTaskScheduler?.startListeningForAppStateUpdates(
+            onEnteringBackground: { [weak self] in self?.handleAppDidEnterBackground() },
+            onEnteringForeground: { [weak self] in self?.handleAppDidBecomeActive() }
+        )
```

Comment:
Shouldn't this happen only if the connection is successful?

Reply:
I think you're right, there is one interesting case when we go background not having connected yet, but current implementation ignores it anyway
```
    private func handleAppDidEnterBackground() {
        // We can't disconnect if we're not connected
        guard connectionStatus == .connected else { return }
    ...
```